src/pretalx/submission/models/submission.py

```diff
@@ -288,38 +297,71 @@ def _set_state(self, new_state, force=False, person=None):
                 )
             )
 
-    def make_submitted(self, person=None, force=False, orga=False):
-        self._set_state(SubmissionStates.SUBMITTED, force, person=person)
+    def update_TalkSlots(self, available_count_old=None):
         from pretalx.schedule.models import TalkSlot
+        logging.debug('update_TalkSlots')
+        # debug: print all now available Talks
+        qs_temp = TalkSlot.objects.filter(
+            submission=self,
+            schedule=self.event.wip_schedule,
+        )
+        logging.debug('TalkSlots old: {}'.format(
+            [i.available_index for i in qs_temp]))
+        # logging.debug('self.state: {}'.format(self.state))
+        if self.state == SubmissionStates.ACCEPTED or \
+            self.state == SubmissionStates.CONFIRMED:
+            # logging.debug('state is ACCEPTED or CONFIRMED')
+            if available_count_old is not None:
+                # logging.debug(
+                #     'available_count_old: {}'.format(available_count_old))
+                # logging.debug(
+                #     'self.available_count {}'.format(self.available_count))
+                # range_to_delete = list(range(
+                #     self.available_count, available_count_old))
+                # logging.debug('range_to_delete: {}'.format(range_to_delete))
+                qs_delete = TalkSlot.objects.filter(
+                    submission=self,
+                    schedule=self.event.wip_schedule,
+                    available_index__in=range_to_delete,
```

Comment:
``range_to_delete`` is not even set here – please add tests for your code to catch non-functional places like this.

Reply:
fixed.
Tests are on the TODO :-)